src/rulesets/oas/functions/oasExample.ts

```diff
@@ -0,0 +1,127 @@
+import type { IFunction, IFunctionContext, IFunctionResult } from '../../../types';
+import type { Dictionary } from '@stoplight/types';
+import type { ISchemaOptions } from '../../../functions/schema';
+
+function isObject(value: unknown): value is Dictionary<any> {
+  return value !== null && typeof value === 'object';
+}
+
+interface IOasExampleOptions {
+  oasVersion: 2 | 3;
+  schemaField: string;
+  type: 'media' | 'schema';
+}
+
+const ORDER_CHECK = {
+  media: {
+    2: [
+      {
+        field: 'examples',
+        multiple: true,
```

Comment:
Media `examples` in OAS2 are bare values, and are usually a MIME type as the name:

```yaml
examples:
  application/json: "the json example"
  application/xml: "the xml example"
```

In OAS3 they have a `value` key:

```yaml
examples:
  foo: 
    value: "the foo example"
  bar:
    value: "the bar example"
```

A more full example:

```
    paths:
      /users:
        post:
          summary: Adds a new user
          requestBody:
            content:
              application/json:     # Media type
                schema:             # Request body contents
                  $ref: '#/components/schemas/User'   # Reference to an object
                examples:    # Child of media type
                  Jessica:   # Example 1
                    value:
                      id: 10
                      name: Jessica Smith
                  Ron:       # Example 2
                    value:
                      id: 11
                      name: Ron Stewart
          responses:
            '200':
              description: OK
```

This example is here. https://swagger.io/docs/specification/adding-examples/

We'll need to make sure we not only say "multiple: true/false" but... is the value keyed behind something or right there?

Reply:
@philsturgeon do we actually have any rule matching these OAS2 media examples?
In case of OAS2 and its examples, we only have the following two rules
```
"oas2-valid-parameter-example": {
      "description": "Examples must be valid against their defined schema.",
      "message": "{{error}}",
      "recommended": true,
      "formats": ["oas2"],
      "severity": 0,
      "type": "validation",
      "given": "$..parameters..[?(@.in == 'body')]..[?(@property !== 'properties' && (@.example || @['x-example']) && (@.type || @.format || @.$ref ))]",
      "then": {
        "function": "oasExample",
        "functionOptions": {
          "schemaField": "$",
          "oasVersion": 2,
          "type": "schema"
        }
      }
    },
    "oas2-valid-definition-example": {
      "description": "Examples must be valid against their defined schema.",
      "message": "{{error}}",
      "recommended": true,
      "formats": ["oas2"],
      "severity": 0,
      "type": "validation",
      "given": "$..definitions..[?(@property !== 'properties' && (@.example || @['x-example']) && (@.type || @.format || @.$ref))]",
      "then": {
        "function": "oasExample",
        "functionOptions": {
          "schemaField": "$",
          "oasVersion": 2,
          "type": "schema"
        }
      }
    }
```
Seems like we might need a new rule?

Reply:
I suppose we will, otherwise we’ll get more complaints that we’re not validating examples.

Reply:
bah, alright.
FWIW, I aimed at supporting oas3 examples (as stated in the PR title) and not oas2, but let's get both done while we are at it. :sweat_smile: